### PR TITLE
Test Message Commitment

### PR DESCRIPTION
**Pull request by MickSandoz**

### CODE REVIEW

File: message_center_compassion/models/gmc_message.py [hunk 1]
```diff
@@ -136,11 +140,11 @@ def update_res_name(self):
...
-        if self.env.context.get('async_mode', True):
-            session = ConnectorSession.from_env(self.env)
-            process_messages_job.delay(session, self._name, self.ids)
-        else:
-            self._process_messages()
+        # if self.env.context.get('async_mode', True):
+        # session = ConnectorSession.from_env(self.env)
+        # process_messages_job.delay(session, self._name, self.ids)
+        # else:
+        self._process_messages()
```
Comment (ecino) [inline message_center_compassion/models/gmc_message.py]: Why did you comment asynchronous processing?

Comment (MickSandoz) [inline message_center_compassion/models/gmc_message.py]: I need to change that, I commented for my tests

File: sponsorship_compassion/mappings/base_sponsorship_mapping.py [hunk 2]
```diff
@@ -24,22 +24,23 @@ class BaseSponsorshipMapping(OnrampMapping):
...
+        "Commitment_ID": "global_id",
+        "CommitmentID": "global_id",
         "CommitmentEndDate": "end_date",
         "CompassNeedKey": ('child_id.local_id', 'compassion.child'),
         "CorrespondenceEndDate": "end_date",
         "CorrespondentCorrespondenceLanguage": None,
         "CorrespondentSupporterGlobalID": None,
         "FinalCommitmentOfLine": None,
-        "HoldID": "reference",
+        "HoldID": ('child_id.hold_id.hold_id', 'compassion.child'),
```
Comment (ecino) [inline sponsorship_compassion/mappings/base_sponsorship_mapping.py]: Should be `compassion.hold` instead of `compassion.child`, no?

Comment (MickSandoz) [inline sponsorship_compassion/mappings/base_sponsorship_mapping.py]: C'est barge, j'ai testé pour les 2 cas (compassion.child ou compassion.hold) et ça marche pour les 2...
Mais je vais mettre avec compassion.hold ça me parrait plus propre ;-)

File: message_center_compassion/models/gmc_message.py [hunk 3]
```diff
@@ -333,11 +333,16 @@ def _send_message(self, message_data):
             # Complete failure (messages were not processed)
             fail = onramp_answer.get('content', {
                 'Error': onramp_answer.get('Error', 'None')})
+            results = onramp_answer.get('content', {}).get(
```
Comment (MickSandoz) [inline message_center_compassion/models/gmc_message.py]: For some message implementation, error message is on content and not in Error...

**Comment by ecino** [comment]
./sponsorship_compassion/models/contracts.py:84:1: W293 blank line contains whitespace

./sponsorship_compassion/models/contracts.py:86:1: W293 blank line contains whitespace

File: message_center_compassion/models/gmc_message.py [hunk 4]
```diff
@@ -333,11 +333,16 @@ def _send_message(self, message_data):
...
+                action.connect_answer_wrapper, [])
+            error_message = str(fail.get('Error', 'None'))
             self.write({
                 'state': 'failure',
                 'failure_reason':
                     '[%s] %s' % (onramp_answer['code'],
-                                 str(fail.get('Error', 'None')))
+                                 error_message if
+                                 error_message != 'None' else
+                                 (results if results else error_message))
```
Comment (ecino) [inline message_center_compassion/models/gmc_message.py]: you can replace by `results or error_message`.
but results is not a string? it should be...

Comment (MickSandoz) [inline message_center_compassion/models/gmc_message.py]: results is not a string, it's a list but it's not a problem to display it in a message.
It's working, I tried

File: child_compassion/models/compassion_hold.py [hunk 5]
```diff
@@ -44,6 +44,7 @@ class CompassionHold(models.Model):
     yield_rate = fields.Float()
     channel = fields.Char()
     source_code = fields.Char()
+    actived = fields.Boolean(default=True, readonly=True)
```
Comment (ecino) [inline child_compassion/models/compassion_hold.py]: `active` should be the name

File: child_compassion/models/compassion_hold.py [hunk 6]
```diff
@@ -60,6 +61,25 @@ def write(self, vals):
...
+        message_vals = {
+            'action_id': action_id,
+            'object_id': self.id
+        }
+
+        if self.child_id.sponsor_id:
+            raise Warning(_("Cancel impossible"), _("This hold is on a "
+                                                    "sponsored child!"))
+        else:
+            self.child_id.actived = False
```
Comment (ecino) [inline child_compassion/models/compassion_hold.py]: The field in child is also `active` not `actived`

File: child_compassion/views/child_holds_view.xml [hunk 7]
```diff
@@ -35,20 +35,38 @@
...
+        <!-- Search view -->
+        <record id="view_child_compassion_filter" model="ir.ui.view">
+            <field name="name">compassion.hold.select</field>
+            <field name="model">compassion.hold</field>
+            <field name="arch" type="xml">
+                <search string="Search hold">
+                    <field name="child_name" filter_domain="[('child_name', 'ilike', self)]"/>
+                    <field name="hold_id"/>
+                    <field name="primary_owner"/>
+                    <filter name="actived" string="Actived" domain="[('actived','=',True)]"/>
```
Comment (ecino) [inline child_compassion/views/child_holds_view.xml]: `Active` is better in English

**Comment by ecino** [comment]
:+1:

**Pull request merged**